Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Internal/Rating.hs

```diff
@@ -0,0 +1,106 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+
+module Domain.Action.Internal.Rating where
+
+import Data.Foldable ()
+import Data.OpenApi (ToSchema)
+import qualified Domain.Types.Person as DP
+import qualified Domain.Types.Rating as DRating
+import Domain.Types.Ride (RideStatus (..))
+import qualified Domain.Types.Ride as DRide
+import qualified EulerHS.Language as L
+import EulerHS.Prelude hiding (id)
+import Kernel.Beam.Functions as B
+import Kernel.Types.APISuccess (APISuccess (Success))
+import Kernel.Types.CacheFlow
+import Kernel.Types.Error
+import Kernel.Types.Id
+import Kernel.Utils.Common hiding (id)
+import SharedLogic.CallBPPInternal (DriverOfferBppInternal)
+import qualified Storage.Queries.Booking as QBooking
+import qualified Storage.Queries.Person as QP
+import qualified Storage.Queries.Rating as QRating
+import qualified Storage.Queries.Ride as QRide
+import Tools.Error (RatingError (..))
+
+data FeedbackReq = FeedbackReq
+  { rideId :: Id DRide.BPPRide,
+    ratingValue :: Int,
+    feedbackDetails :: Maybe Text
+  }
+  deriving (Show, Generic, ToJSON, FromJSON, ToSchema)
+
+rating ::
+  ( MonadFlow m,
+    EsqDBFlow m r,
+    CacheFlow m r,
+    EncFlow m r,
+    HasFlowEnv m r '["minimumDriverRatesCount" ::: Int],
+    HasFlowEnv m r '["driverOfferBppInternal" ::: DriverOfferBppInternal]
+  ) =>
+  Maybe Text ->
+  FeedbackReq ->
+  m APISuccess
+rating apiKey FeedbackReq {..} = do
+  driverOfferBppInternal <- asks (.driverOfferBppInternal)
```

Comment:
Don't use this whole structure,  define internalAPIKey separately in Config and enviornment and use it.